src/Resolver/Asset/ResourceAssetResolverChain.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\GraphQL\Search\Resolver\Asset;
+
+use Atoolo\GraphQL\Search\Types\Asset;
+use Atoolo\GraphQL\Search\Types\SymbolicImage;
+use Atoolo\Resource\Resource;
+use Overblog\GraphQLBundle\Definition\ArgumentInterface;
+
+class ResourceAssetResolverChain implements ResourceAssetResolver, ResourceSymbolicImageResolver
+{
+    /**
+     * @param array<ResourceAssetResolver> $resolvers
+     */
+    public function __construct(
+        private readonly iterable $resolvers
+    ) {
+    }
+
+    public function getAsset(
+        Resource $resource,
+        ArgumentInterface $args
+    ): ?Asset {
+        foreach ($this->resolvers as $resolver) {
+            $asset = $resolver->getAsset(
+                $resource,
+                $args
+            );
+            if ($asset !== null) {
+                return $asset;
+            }
+        }
+        return null;
+    }
+
+    public function getSymbolicImage(
+        Resource $resource,
+        ArgumentInterface $args
+    ): ?SymbolicImage {
+        foreach ($this->resolvers as $resolver) {
+            if (!$resolver instanceof ResourceSymbolicImageResolver) {
+                continue;
+            }
+            $symbolicImage = $resolver->getSymbolicImage(
+                $resource,
+                $args
+            );
+            if ($symbolicImage !== null) {
+                return $symbolicImage;
+            }
+        }
+        return null;
+    }
```

Comment:
The chain no longer needs a `getSymbolicImage()`.

Reply:
It's intended to have both a `getAsset` and a `getSymbolicImage` function, because I want to be able to specifically resolve a SymbolicImage in some cases. Wouldn't be possible if there was only a general purpose `getAsset` function.

Reply:
But then there is the `ResourceSymbolicImageResolver`. So it's all very confusing.